full-moon/src/ast/parsers.rs

```diff
@@ -2437,21 +2449,62 @@ fn parse_simple_type(
 #[cfg(feature = "luau")]
 fn parse_type_suffix(
     state: &mut ParserState,
-    simple_type: ast::TypeInfo,
+    simple_type: Option<ast::TypeInfo>,
 ) -> ParserResult<ast::TypeInfo> {
     let mut is_union = false;
     let mut is_intersection = false;
 
+    let mut types = Punctuated::new();
+
+    let leading = if simple_type.is_some() {
```

Comment:
`or_else` probably better

Reply:
I'm not sure what I would use `or_else` on here

Reply:
Ah I see--it is inverted